### ISSUE TITLE
Comments are treated oddly by the heex parser.

### ISSUE THREAD

**Issue by JonRowe** [phoenixframework/phoenix_live_view#1590]
### Environment

* Elixir version (elixir -v): 1.12.2 / OTP 24
* Phoenix version (mix deps): 1.5.12
* Phoenix LiveView version (mix deps): 0.16.1
* Operating system: OS X Catalina
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome, but its a compiler error.
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes, its a error during compilation.

### Actual behavior

Using comments as:
```
<!-- { some_value } -->
```

Leads to a literal:
```
<!-- { some_value } -->
```

In the source output, yet:

```
<!-- <%= some_value %> -->
```

Leads to a compile time error of:

```
 expected closing `-->` for comment
```

### Expected behaviour

One or the other forms of syntax should be valid (the former would be least suprising) so that dynamic values can be output into comments.

**Comment by josevalim** [comment]
Yes, you cannot use EEx inside ANY html tag. Not even comments. Only in the body. I will improve the error message here. :)

**Comment by JonRowe** [comment]
@josevalim the other problem is that `{ ... }` syntax doesn't work here it doesn't get evaluated

**Comment by josevalim** [comment]
@JonRowe hrm... do you want to interpolate inside a comment? :)

**Comment by JonRowe** [comment]
The comments are used here to delineate sections of a diagram for developer and test sanity, so yes I do 😬

**Comment by josevalim** [comment]
@JonRowe and you want whatever this is to be sent to the client and be in the final HTML?

**Comment by JonRowe** [comment]
e.g.
```
<!-- 4.0 -->
<g>
  <path class="line" d="M 13.408493,-12.9484320 C 14.613517114834526,-12.602 <snip>"></path>
</g>
<!-- 6.0 -->
<g>
  <path class="line" d="M 13.408493,-12.9484320 C 14.613517114834526,-12.602 <snip>"></path>
</g>    
```

**Comment by JonRowe** [comment]
Yes I do, this worked before with `leex` and I'm currently resorting to `<%= raw("<!-- #{ value } -->") %>`

**Comment by josevalim** [comment]
Ack. So I guess we need to discuss this and see if we allow regular `{}` inside comment tags too. It seems that React and Vue don't support it either and you need to do something similar to your raw workaround.

**Comment by chrismccord** [comment]
This may be what you already have, but a function component works really nicely as a workaround:

```eex
  <.comment>
    This is a comment <%= @foo %>
  </.comment>
```

```elixir
  def comment(assigns) do
    ~H"""
    <%= raw("<!--") %><%= render_block(@inner_block) %><%= raw("-->") %>
    """
  end
```

**Comment by chrismccord** [comment]
could also have a shorthand form for one-liners, `<.comment text={value} />`

**Comment by JonRowe** [comment]
I'm doing:
```
<%= comment(value) %>
```
with:
```
defp comment(string) do
  case html_escape(string) do
    {:safe, string} -> raw("<!-- #{string} -->")
    _ -> ""
  end
end
```

**Comment by chrismccord** [comment]
Nice. The function component approach will allow a full block of content so for multiline stuff it will be really nice. I think this kind of thing covers all cases, so I'm willing to punt on this for now.

**Comment by tadasajon** [comment]
>  You could accomplish this by wrapping your desired template content in a function component 

Thanks. This is perfect and helpful information. I'm going to work on figuring this out immediately.

But wouldn't it make sense for Phoenix/Heex to come with this `<.ignore>` function component built in?  Otherwise every newbie Phoenix developer is going to have to reimplement this functionality themselves in their code base.

**Comment by josevalim** [comment]
I made it so `<!-- <%= ... %> -->` works, but it will render the contents inside and send it to the client (which is how it worked with EEx in the first place).

**Comment by JonRowe** [comment]
❤️  🧡  💛  💚  💜

**Comment by DavidRawk-Blake** [comment]
Just for fun I tried

`<%= raw("<!-- #{ "asdfasd" } -->") %>` 
and I get this

`
<span data-phx-component="4"> asdfasd </span>
`
